apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/CommandSerializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.skywalking.apm.network.trace.component.command;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.skywalking.apm.network.common.Command;
+
+/**
+ *
+ * @author Zhang Xin
+ */
+public class CommandSerializer {
+
+    public final static Map<String, Deserializable> COMMAND_SERIALIZE_MAPPING = new HashMap<String, Deserializable>(5);
+
+    static {
+        COMMAND_SERIALIZE_MAPPING.put("EndpointMetadataReset", new EndpointResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("InstanceMetadataReset", new InstanceResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("NetworkAddressMetadataReset", new NetworkResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("ServiceMetadataReset", new ServiceResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("TraceIgnore", new TraceIgnoreCommand("FOR_DESERIALIZE"));
```

Comment:
I am totally confused about these. All command construct arguments say `serialNumber`. What are you trying to do here?

I don't think you should use this construct. @peng-yongsheng suggestion?